pkg/stub/stub.go

```diff
@@ -764,3 +767,129 @@ func (stub *stub) setupHandlers() error {
 
 	return nil
 }
+
+type origConfigureInterface interface {
```

Comment:
deprecations?

Reply:
Do you mean that I should make these old interfaces all exported (OrigConfigureInterface, etc) and mark them deprecated in their docstring ?

Reply:
I was reading through the code trying to gronk what/if any plan there was for deprecating the pre-context injected versions of the api

- new release, no deprecations
- new api version, with deprecations
- ...

Reply:
I mean I can see the code rejects.. :-)

Reply:
> I mean I can see the code rejects old handlers.. :-)

Ah, ok that. I think we haven't declared the APIs stable yet. So I think in principle it should be still okay to make API breaking changes. We kept those old interfaces as unexported ones prefixed with `orig`, just to be able to test for them and bail out with an error message if a plugin implements old interfaces.

I don't know what would be the best option here. 1) Don't test for old interfaces and let plugin creation fail with something like `plugin XYZ does not implement any NRI request handlers`. 2) Do what we do now. 3) Mark the interfaces deprecated and still accept/handle plugins that implement old interfaces. Is that extra trouble worth it at this point, yet ?

BTW, once we merge this I'll need to go and file another update to address the missing resources parameter from UpdateContainers, which is another API-breaking change.

Reply:
Okay. Based on our discussion, I ditched all the tests for old interfaces. Plugins anyway need to be updated and if they are not, the will fail to start up with a `plugin XYZ does not implement any NRI request handlers` error message.

I updated the documentation with a note about the current lack of guaranteed API stability before v1.0.0. I gave a guesstimate for a v1.0.0 release. Finally, I added a brief release notes file which lists the changes so far between each tagged release, and for the pending untagged v0.4.0.